### ISSUE TITLE
linter: `react/react-compiler` misses the "value blocks within a try/catch" bailout

### ISSUE THREAD

**Issue by madebyjulz** [oxc-project/oxc#25342]
### What version of Oxlint are you using?

1.77.0

### What command did you run?

`oxlint --config .oxlintrc.json repro.tsx`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["react"],
  "rules": {
    "react/react-compiler": ["error", { "reportAllBailouts": true }]
  }
}
```

### What happened?

Optional chaining (or `&&` / `||` / `??` / a ternary) inside a `try` block makes
babel-plugin-react-compiler bail out on the whole component. oxlint doesn't report it,
even with `reportAllBailouts: true`, so the component looks fine and silently loses
memoization once the compiler is turned on.

The neighbouring bailouts in the same `try` statement *are* reported, so I think this is
just one check that didn't make it into the port rather than anything to do with config
or severity.

**repro.tsx**

```tsx
import {useState} from 'react';

export function Component({user}: {user: {id: string} | null}) {
  const [current, setCurrent] = useState<string | null>(null);

  const onScan = async (id: string) => {
    try {
      if (user?.id !== id) {
        setCurrent(await Promise.resolve(id));
      }
    } catch {
      setCurrent(null);
    }
  };

  return <button onClick={() => onScan('a')}>{current}</button>;
}
```

`oxlint --config .oxlintrc.json repro.tsx` → exit 0, nothing reported.

babel-plugin-react-compiler@1.0.0 on the same file:

```
CompileError — Support value blocks (conditional, logical, optional chaining, etc) within a try/catch statement
```

Sanity check that the rule was actually running: add a `finally` to that same `try` and
oxlint reports it as expected.

```
control.tsx:7:5: error react(react-compiler): Todo: (BuildHIR::lowerStatement) Handle TryStatement with a finalizer ('finally') clause
```

**Where it lives upstream**

It isn't in `BuildHIR.ts` — it fires later, on the `maybe-throw` terminal, which may be why
it got missed. Present in every published version I checked, 1.0.0 included.

In `react_compiler_lowering/build_hir.rs` the `TryStatement` Todos for a missing catch
clause and for a finalizer are both there, and `resolve_throw_handler()` only has the one
caller for `ThrowStatement inside of try/catch`. I couldn't find a counterpart for the
`maybe-throw` check — though possibly oxc's HIR just doesn't model those terminals the
same way.

**Comment by madebyjulz** [comment]
Closing — refiling through the linter bug report template so it gets triaged properly.
